### PR TITLE
Improve content form ux [part 1]

### PR DESCRIPTION
**Pull request by thiagomurakami**
#### What is the purpose of this pull request?
- [x] Add debounce to highlight handler
- [x] Separate layout and content schema at the root.
- [x] New rule to if form data was modified.
- [x] Change "reset" logic on content.

#### What problem is this solving?
1. Annoying modal alert every time you change something, even when nothing has changed.
2. Avoid iframe excessive scrolling when hovering the list.

#### How should this be manually tested?
[Workspace](https://contentformux--storecomponents.myvtexdev.com/admin/cms/storefront)

#### Types of changes
- [x] Bug fix (a non-breaking change which fixes an issue)

### CODE REVIEW

File: react/components/EditorContainer/Sidebar/ComponentEditor/index.tsx [hunk 1]
```diff
@@ -46,29 +81,21 @@ const ComponentEditor: React.FunctionComponent<Props> = ({
...
-      : null
-
-  const componentSchema = getComponentSchema({
-    component: componentImplementation,
-    contentSchema,
-    propsOrContent: extension[isContent ? 'content' : 'props'],
-    runtime: iframeRuntime,
-  })
+  const { componentSchema, uiSchema } = useMemo(
+    () => getSchemas({ contentSchema, editTreePath, iframeRuntime, isContent }),
```
Comment (alcar) [inline react/components/EditorContainer/Sidebar/ComponentEditor/index.tsx]: I think we should (eventually) remove the `isContent` prop, and use:
```ts
const isContent = mode === 'content'
```

File: react/components/EditorContainer/Sidebar/ComponentEditor/index.tsx [hunk 2]
```diff
@@ -46,29 +81,21 @@ const ComponentEditor: React.FunctionComponent<Props> = ({
...
+    [editTreePath]
```
Comment (alcar) [inline react/components/EditorContainer/Sidebar/ComponentEditor/index.tsx]: I think `mode` should also be a "watched" variable.

File: react/components/EditorContainer/Sidebar/ConfigurationList/index.tsx [hunk 3]
```diff
@@ -70,13 +71,40 @@ defineMessages({
...
+const omitUndefined = pickBy(val => typeof val !== 'undefined')
+
 class ConfigurationList extends React.Component<Props, State> {
   private component: Extension['component']
   private componentImplementation: RenderComponent<any, any> | null
   private componentProperties: ComponentSchema['properties']
   private componentTitle: ComponentSchema['title']
   private contentSchema: JSONSchema6
 
+  private handleFormChange: FormProps<object>['onChange'] = debounce(event => {
```
Comment (alcar) [inline react/components/EditorContainer/Sidebar/ConfigurationList/index.tsx]: :pick: Why is a private method being defined before the constructor?

File: react/components/EditorContainer/Sidebar/ConfigurationList/index.tsx [hunk 4]
```diff
@@ -339,8 +348,16 @@ class ConfigurationList extends React.Component<Props, State> {
   }
 
   private handleConfigurationDiscard = () => {
-    this.props.formMeta.setWasModified(false, () => {
+    const { editor, iframeRuntime } = this.props
+
+    this.props.formMeta.setWasModified(false, async () => {
```
Comment (alcar) [inline react/components/EditorContainer/Sidebar/ConfigurationList/index.tsx]: The second argument of `setWasModified` could be replaced with a call to `handleContentBack`.

File: react/components/EditorContainer/Sidebar/ConfigurationList/index.tsx (hunk 3 above)
Comment (thiagomurakami) [reply react/components/EditorContainer/Sidebar/ConfigurationList/index.tsx]: Because it is a variable :|

**Pull request merged**